packages/cursorless-vscode/src/ide/vscode/VscodeFileSystem.ts

```diff
@@ -1,26 +1,90 @@
-import { Disposable, FileSystem, PathChangeListener } from "@cursorless/common";
-import { RelativePattern, workspace } from "vscode";
-import * as path from "path";
+import {
+  Disposable,
+  FileSystem,
+  PathChangeListener,
+  RunMode,
+} from "@cursorless/common";
+import { isAbsolute } from "path";
+import { TextDecoder } from "util";
+import * as vscode from "vscode";
 
 export class VscodeFileSystem implements FileSystem {
   public readonly cursorlessTalonStateJsonPath: string;
   public readonly cursorlessCommandHistoryDirPath: string;
+  public readonly cursorlessDir: string;
 
-  constructor(public readonly cursorlessDir: string) {
-    this.cursorlessTalonStateJsonPath = path.join(
+  private decoder: TextDecoder = new TextDecoder("utf-8");
+
+  constructor(
+    private readonly extensionContext: vscode.ExtensionContext,
+    private readonly runMode: RunMode,
+    cursorlessDirPath: string,
+    cursorlessDirName: string,
+  ) {
+    this.cursorlessDir = this.join(cursorlessDirPath, cursorlessDirName);
+    this.cursorlessTalonStateJsonPath = this.join(
       this.cursorlessDir,
       "state.json",
     );
-    this.cursorlessCommandHistoryDirPath = path.join(
+    this.cursorlessCommandHistoryDirPath = this.join(
       this.cursorlessDir,
       "commandHistory",
     );
   }
 
-  watchDir(path: string, onDidChange: PathChangeListener): Disposable {
+  public async initialize(): Promise<void> {
+    await vscode.workspace.fs.createDirectory(
+      vscode.Uri.file(this.cursorlessDir),
+    );
+  }
+
+  public join(path1: string, path2: string): string {
+    // FIXME: Look into whether we can use joinPath(...).path. The
+    // question is whether Uri will correctly remove the Uri.base's path
+    // from the returned path.
+    return `${path1}/${path2}`;
```

Comment:
yeah we should try to do a proper join

Reply:
I confirmed that we can't use vscode.Uri.joinPath and then extract the path, as I said we should look into, because that will lose the extension's base path if there is one.  Since we also can't import `path` in the web version, and since vscore.Uri always uses forward-slash (see comment elsewhere), I think an array join on "/" is sufficient here.

Reply:
I've gone back to path.join, so this is no longer relevant.